rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -2973,6 +2994,14 @@ RMW_Connext_Service::send_response(
   rr_msg.gid.implementation_identifier = RMW_CONNEXTDDS_ID;
   rr_msg.payload = const_cast<void *>(ros_response);
 
+  DDS_Time_t timestamp;
+  if (DDS_RETCODE_OK !=
+    DDS_DomainParticipant_get_current_time(this->reply_pub->dds_participant(), &timestamp))
+  {
+    RMW_CONNEXT_LOG_ERROR_SET("failed to write message to DDS")
```

Comment:
I would change the message to be a bit more specific to this failure, e.g.

```suggestion
    RMW_CONNEXT_LOG_ERROR_SET("failed to get current time from DDS DomainParticipant")
```